coder/coder
# fix: ui: workspace bumpers now honour template max_ttl

- chore: WorkspacePage: invert workspace schedule bumper logic for readibility
- fix: ui: workspace bumpers now honour template max_ttl
- chore: refactor workspace schedule bumper logic mostly into `util/schedule.ts` and unit test separately

Verification Steps:
- Create and start a workspace from a template that has a max_ttl of less than 24 hours and has autostop enabled.
- Repeatedly click the ➕ icon by the schedule -> it should not allow extending the workspace deadline past template max_ttl
- Repeatedly click the ➖  icon by the schedule -> it should not allow decreasing the workspace deadline before now plus 30 minutes

Fixes https://github.com/coder/coder/issues/2764.
